packages/telemetry/test/index.spec.ts

```diff
@@ -0,0 +1,297 @@
+import { expect } from 'chai'
+
+import { Telemetry } from '../src'
+
+import { NodeTracerProvider } from '@opentelemetry/sdk-trace-node'
+import { BatchSpanProcessor } from '@opentelemetry/sdk-trace-base'
+
+import { OTLPTraceExporter as OTLPTraceExporterCloud } from '../src/span-exporters/cloud-span-exporter'
+
+describe('init', () => {
+  it('creates a new instance', () => {
+    const exporter = new OTLPTraceExporterCloud()
+
+    const tel = new Telemetry({
+      namespace: 'namespace',
+      Provider: NodeTracerProvider,
+      detectors: [],
+      exporter,
+      version: 'version',
+      SpanProcessor: BatchSpanProcessor,
+    })
+
+    expect(tel).to.not.be.undefined
+    expect(tel.provider).is.instanceOf(NodeTracerProvider)
+    expect(tel.provider.resource.attributes['service.namespace']).to.equal('namespace')
+    expect(tel.provider.resource.attributes['service.version']).to.equal('version')
+    expect(tel.provider.resource.attributes['service.name']).to.equal('cypress-app')
+    // @ts-ignore
+    expect(tel.provider.activeSpanProcessor._spanProcessors[0]).is.instanceOf(BatchSpanProcessor)
+    expect(tel.getExporter()).to.equal(exporter)
+    expect(tel.rootContext).to.be.undefined
+  })
+
+  it('creates a new instance', () => {
+    const exporter = new OTLPTraceExporterCloud()
+
+    const tel = new Telemetry({
+      namespace: 'namespace',
+      Provider: NodeTracerProvider,
+      detectors: [],
+      exporter,
+      version: 'version',
+      rootContextObject: { traceparent: '00-a14c8519972996a2a0748f2c8db5a775-4ad8bd26672a01b0-01' },
+      SpanProcessor: BatchSpanProcessor,
+    })
+
+    expect(tel).to.not.be.undefined
+    expect(tel.rootContext).to.not.be.undefined
+  })
+})
+
+describe('startSpan', () => {
+  it('starts a span with an external parent id', () => {
+    const exporter = new OTLPTraceExporterCloud()
+
+    const tel = new Telemetry({
+      namespace: 'namespace',
+      Provider: NodeTracerProvider,
+      detectors: [],
+      exporter,
+      version: 'version',
+      rootContextObject: { traceparent: '00-a14c8519972996a2a0748f2c8db5a775-4ad8bd26672a01b0-01' },
+      SpanProcessor: BatchSpanProcessor,
+    })
+
+    const span = tel.startSpan({ name: 'span' })
+
+    // @ts-ignore
```

Comment:
I'd make these `ts-expect-error` in the case we fix the type, the tests will fail and we can remove the `ts-expect-error` statements

Reply:
these are likely never going to be 'fixed'. we're testing 'private' properties on open telemetry objects